src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -142,6 +148,21 @@ async def collect_thirdparty_modules(
                 continue
             applicable_lockfile_entries[entry] = ct
 
+            artifact_requirement_source = dataclasses.replace(
+                artifact_requirement,
+                coordinate=dataclasses.replace(
+                    artifact_requirement.coordinate, classifier="sources"
```

Comment:
I recall from the PR description, that you had some concern about hard-coding to "sources" classifier. Seems fine to me. I suggest defining a set of constants for well-known classifier names though and using a constant here instead of the open-coded `"sources"` string.